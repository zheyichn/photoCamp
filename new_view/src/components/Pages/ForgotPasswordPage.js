import React from "react";
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import Divider from '@mui/material/Divider';
import Link from '@mui/material/Link';
import {ThemeProvider} from '@mui/material/styles';
import ThemeBuilder from '../../themes/ThemeBuilder';
import Header from '../Activity/Header';

const ForgotPasswordTextField = (props) => {
    const myProp = props;
    const label = myProp.label;
    return(
        <TextField
          label={label}
          variant="outlined"
          size="small"
          sx={{
            input: { background: '#FFFFFF'},
          }}
        />
    );
  };

export default function ForgotPasswordPage() {

    return (
        <>
            <Header />
            <ThemeProvider theme={ThemeBuilder()}>
                <Grid
                    container
                    spacing={0}
                    direction="row"
                    alignItems="center"
                    justifyContent="center"
                    style={{
                        minHeight: '100vh',
                        gap: '20px',
                    }}
                >
                    <Grid
                        item
                        xs={5}
                        backgroundColor='#D9D9D9'
                        borderRadius='12px'
                        style={{
                            minHeight: '300px',
                            minWidth: '300px',
                        }}
                    >
                        <Box
                            marginX={6}
                            marginY={2}
                            sx={{
                                display: 'flex',
                                flexDirection: 'column',
                                justifyContent: 'center',
                                alignContent: 'center',
                                gap: '15px'
                            }}
                        >
                            <Typography align='center' style={{fontSize: 24}}>
                                Forgot Password?
                            </Typography>
                            <Typography align='left' style={{fontSize: 18}}>
                                We’ve got your back. Enter your email or
                                username and we’ll send you a link to reset your password.
                            </Typography>
                            <ForgotPasswordTextField label="Email / Username" />
                            <Button
                                variant="contained"
                                size="small"
                                sx={{width: '140px'}}
                            >
                                Send me the link!
                            </Button>
                            <Divider sx={{ backgroundColor: '#000000' }}/>
                            <Typography align="center" style={{fontSize: 12}}>
                                <Link href="/CreateNewAccount">
                                    Create New Account
                                </Link>
                            </Typography>
                        </Box>
                    </Grid>
                </Grid>     
            </ThemeProvider>
        </>
    )

}