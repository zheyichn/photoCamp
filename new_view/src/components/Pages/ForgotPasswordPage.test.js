import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import ForgotPasswordPage from "./ForgotPasswordPage";

// afterEach function runs after each test suite is executed
afterEach(() => {
    cleanup();
});

describe("Forgot Password Page", () => {
  
    // Test1: match snapshots
    it("Page renders correctly", () => {
      const tree = renderer
        .create(
          <ForgotPasswordPage />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});