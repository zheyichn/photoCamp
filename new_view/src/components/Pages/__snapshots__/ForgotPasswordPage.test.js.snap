// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password Page Page renders correctly 1`] = `
[
  <div>
    <div
      className="MuiGrid-root css-54nrqh-MuiGrid-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
          style={
            {
              "backgroundColor": "white",
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item css-fom513-MuiGrid-root"
            style={
              {
                "backgroundColor": "white",
              }
            }
          >
            <div
              className="MuiTypography-root MuiTypography-h6 css-fd7qc2-MuiTypography-root"
            >
              PhotoCamp
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container css-154w0yo-MuiGrid-root"
    style={
      {
        "gap": "20px",
        "minHeight": "100vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-1nnzfnh-MuiGrid-root"
      style={
        {
          "minHeight": "300px",
          "minWidth": "300px",
        }
      }
    >
      <div
        className="MuiBox-root css-g67067"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-3mddyt-MuiTypography-root"
          style={
            {
              "fontSize": 24,
            }
          }
        >
          Forgot Password?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-4rnw4b-MuiTypography-root"
          style={
            {
              "fontSize": 18,
            }
          }
        >
          We’ve got your back. Enter your email or username and we’ll send you a link to reset your password.
        </p>
        <div
          className="MuiFormControl-root MuiTextField-root css-4mufqi-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-13tblej-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor=":r0:"
            id=":r0:-label"
          >
            Email / Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-175sm4b-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id=":r0:"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Email / Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-1lk5wtx-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Send me the link!
        </button>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-ogpflq-MuiDivider-root"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-3mddyt-MuiTypography-root"
          style={
            {
              "fontSize": 12,
            }
          }
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="/CreateNewAccount"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Create New Account
          </a>
        </p>
      </div>
    </div>
  </div>,
]
`;
